### PR TITLE
Added wifi based geolocation to base libraries

### PR DESCRIPTION
**Pull request by alistairuk**
Added a library geolocation using wifi triangulation to base libraries. The back end will need moving to the EMF Camp server at some point as but it will work with Alistair's server and API keys for now.

### CODE REVIEW

File: geo.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			geojson += ","
+		else:
+			firsttime=False
+		# get the bssid
+		bssid = ''
+		for b in i['bssid']:
+			bssid += "%02x" % (b);
+		# Write the signal to the cache file
+		geojson += "\""+bssid+"\":"+str(-i['rssi'])
+	geojson += "}"
```
Comment (marekventur) [inline geo.py]: hand-writing a json like this is probably not a good idea, you can use ```ujson.dumps()``` instead

**Comment by marekventur** [comment]
This is very cool, thank you! 

However, I'm not sure we want to put it in the base set of libraries:
* No app is including it at the moment, so it gets bootstrapped onto badges and then not used (which could change soon, of course)
* It's connecting to a special server endpoint that has to be maintained. While this is fine while it works, it would be quite confusing when it breaks (and you're on holiday for example)
* It's quite short, so apps that need it could include the file itself and there are not much saving to be had by sharing it between apps.

Can I suggest an alternative? You could create a repo with the lib, the server code, some instructions and maybe a sample app and we can link it on the wiki and people can include the lib file themselves as part of their app. We could also put a file into the examples folder.

What's the view of the rest of the team?

**Comment by Jonty** [comment]
@alistairuk has emailed me a zip with all the server-side stuff, so we can run it ourselves if we want to.

**Comment by marekventur** [comment]
Oh, sorry, didn't know there was some talk outside github. I'll hand the thread over to someone who has kept up with IRC then ;)

**Comment by marekventur** [comment]
btw, could you move the file into the lib folder? it looks like it's in the root atm.

**Comment by alistairuk** [comment]
Da, that is me being an idiot. I have moved it in the patch branch and will make another pull request. I have also used ujson as you suggested. Thanks for the pointer. Any more like that are most welcome.

**Comment by alistairuk** [comment]
Tidying this up. No longer relevant.

**Pull request closed without merging**